### DISCUSSION TITLE
add Sparrow (Terminal)

### DISCUSSION THREAD

**Discussion by vv01f** [raspiblitz/raspiblitz#3915] [Feature / App Ideas]
s.a. https://github.com/sparrowwallet/sparrow/releases/tag/1.7.0

maybe brings whirlpool to raspiblitz?

**Comment by openoms** [comment]
YES!

```
# see https://sparrowwallet.com/download/ for the latest
VERSION=1.7.6
# for RPi: 1_arm64 # for PC: 1_amd64
CPU=1_arm64
# download
wget -O sparrow-server_$VERSION-$CPU.deb https://github.com/sparrowwallet/sparrow/releases/download/$VERSION/sparrow-server_$VERSION-$CPU.deb
wget -O sparrow-$VERSION-manifest.txt https://github.com/sparrowwallet/sparrow/releases/download/$VERSION/sparrow-$VERSION-manifest.txt
wget -O sparrow-$VERSION-manifest.txt.asc https://github.com/sparrowwallet/sparrow/releases/download/$VERSION/sparrow-$VERSION-manifest.txt.asc
# checksum check - look for 'OK'
sha256sum -c sparrow-$VERSION-manifest.txt --ignore-missing
# get PGP pubkey
curl https://keybase.io/craigraw/pgp_keys.asc | gpg --import
# signature check - look for Good signature from "Craig Raw <craigraw@gmail.com>"
gpg --verify sparrow-$VERSION-manifest.txt.asc

# install
sudo dpkg -i sparrow-server_$VERSION-$CPU.deb

# options
/opt/sparrow/bin/Sparrow --help

# start
/opt/sparrow/bin/Sparrow
```
![image](https://user-images.githubusercontent.com/43343391/198314561-a31c1a94-962f-4a0e-b251-c540b5cfa887.png)


The config file is in the home folder of the user started Sparrow (default is `admin`):
```
/home/admin/.sparrow/config
```

Example after the first run with the local Tor proxy set:
```
cat .sparrow/config
{
  "mode": "ONLINE",
  "loadRecentWallets": true,
  "validateDerivationPaths": true,
  "groupByAddress": true,
  "includeMempoolOutputs": true,
  "notifyNewTransactions": true,
  "checkNewVersions": true,
  "openWalletsInNewWindows": false,
  "hideEmptyUsedAddresses": false,
  "showTransactionHex": true,
  "showLoadingLog": true,
  "showAddressTransactionCount": false,
  "showDeprecatedImportExport": false,
  "preventSleep": true,
  "dustAttackThreshold": 1000,
  "enumerateHwPeriod": 30,
  "serverType": "ELECTRUM_SERVER",
  "electrumServer": "tcp://192.168.3.170:50021",
  "recentElectrumServers": [
    "tcp://192.168.3.170:50021"
  ],
  "useProxy": true,
  "proxyServer": "127.0.0.1:9050",
  "maxServerTimeout": 34,
  "maxPageSize": 100,
  "usePayNym": false,
  "sameAppMixing": false
}
```
EDIT:
added to config:` "preventSleep": true,`
added VERSION and CPU variables

**Reply by marc3linho** [reply]
Settings to connect to Electrum Rust Server:
Private Electrum─────────────────────────────┬───────────────────────┐    
   │ URL               127.0.0.1                   │▼  50001               │    
   │ Alias (optional)                                                      │    
   │                                                                       │    
   │ Use SSL?          No  │▼                                              │    
   │ Certificate                                       <Select...>         │    
   │                                                                       │    
   <del>│ Use Proxy?        Yes │▼                                              │</del>    
   <del>│ Proxy URL         127.0.0.1                       9050 </del>              
Edit: 
change due to openoms comment below

**Reply by openoms** [reply]
> Settings to connect to Electrum Rust Server:
> Private Electrum─────────────────────────────┬───────────────────────┐    
>    │ URL               127.0.0.1                   │▼  50001               │    
>    │ Alias (optional)                                                      │    
>    │                                                                       │    
>    │ Use SSL?          No  │▼                                              │    
>    │ Certificate                                       <Select...>         │    
>    │                                                                       │    
>    │ Use Proxy?        Yes │▼                                              │    
>    │ Proxy URL         127.0.0.1                       9050               

don't use the Tor proxy to connect to the localhost.
You can't tavel out to the internet and connect back locally. It must be direct (no proxy).

**Reply by marc3linho** [reply]
now that i read it with your explanation, you are absolutely right! 
thank you

**Comment by Sep21-2009** [comment]
Change 
"preventSleep": false,
to 
"preventSleep": true,

**Comment by openoms** [comment]
A guide with screenshots and using tmux: https://sovrnbitcoiner.com/an-iphone-enters-the-poolside-chat/

**Reply by marc3linho** [reply]
The link seems to be dead

**Comment by openoms** [comment]
Update for amd64:
```
# see https://sparrowwallet.com/download/ for the latest
VERSION=2.0.0
# for RPi: 1_arm64 # for PC: 1_amd64
CPU=1_amd64
# download
wget -O sparrow-server_$VERSION-$CPU.deb https://github.com/sparrowwallet/sparrow/releases/download/$VERSION/sparrow-server_$VERSION-$CPU.deb
wget -O sparrow-$VERSION-manifest.txt https://github.com/sparrowwallet/sparrow/releases/download/$VERSION/sparrow-$VERSION-manifest.txt
wget -O sparrow-$VERSION-manifest.txt.asc https://github.com/sparrowwallet/sparrow/releases/download/$VERSION/sparrow-$VERSION-manifest.txt.asc
# checksum check - look for 'OK'
sha256sum -c sparrow-$VERSION-manifest.txt --ignore-missing
# get PGP pubkey
curl https://keybase.io/craigraw/pgp_keys.asc | gpg --import
# signature check - look for Good signature from "Craig Raw <craigraw@gmail.com>"
gpg --verify sparrow-$VERSION-manifest.txt.asc

# install
sudo dpkg -i sparrow-server_$VERSION-$CPU.deb

# options
/opt/sparrow/bin/Sparrow --help

# start
/opt/sparrow/bin/Sparrow
```
